app/scripts/lib/transaction/util.ts

```diff
@@ -237,13 +237,17 @@ async function addTransactionOrUserOperation(
 async function addTransactionWithController(
   request: FinalAddTransactionRequest,
 ) {
-  const { transactionController, transactionOptions, transactionParams } =
-    request;
+  const {
+    transactionController,
+    transactionOptions,
+    transactionParams,
+    networkClientId,
+  } = request;
   const { result, transactionMeta } =
-    await transactionController.addTransaction(
-      transactionParams,
-      transactionOptions,
-    );
+    await transactionController.addTransaction(transactionParams, {
+      ...transactionOptions,
+      ...(process.env.TRANSACTION_MULTICHAIN ? { networkClientId } : {}),
```

Comment:
> If we're wanting to support the multi-chain flag, do we also need to read the networkClientId from the dappRequest in addDappTransaction, assuming it's available in the middleware?

Yeah, good point. I've added this to `getAddTransactionRequest`.

> Currently I think this is fixed to this.networkController.state.selectedNetworkClientId via getAddTransactionRequest in MetamaskController, meaning it will only ever be the global network so no different to not providing a network client ID?

Right, for transactions added via wallet ui, there is no dappRequest and it can currently be assumed the networkClientId for those transactions is the `selectedNetworkClientId` at the time of add. Because of this, we still want to fallback to `selectedNetworkClientId` in `getAddTransactionRequest` for transactions originating from the wallet ui.

> Or does TRANSACTION_MULTICHAIN just mean we're using the controller in multichain mode but still relying on the global network?

TRANSACTION_MULTICHAIN flag enabled means that the controller itself will be capable of processing transactions for all networkClientIds as well as the globally selected network. However since we also ensure that networkClientId is passed in where appropriate on the extension side when the flag is on, it means that the multichain flow should be the only flow being used (except for a few places like swaps)